typing/typedecl.ml

```diff
@@ -1227,7 +1227,13 @@ let transl_type_decl env rec_flag sdecl_list =
     (fun sdecl (tdecl, _shape) ->
       let decl = tdecl.typ_type in
        match Ctype.closed_type_decl decl with
-         Some ty -> raise(Error(sdecl.ptype_loc, Unbound_type_var(ty,decl)))
+         Some ty ->
+           let err = Error(sdecl.ptype_loc, Unbound_type_var(ty,decl)) in
+           if !Clflags.typing_recovery
+           && (not (Typing_recovery.erroneous_type_check ty))
+           then raise(err) (* Strange but if typing recovery is enabled,
+                              it should only if erroneous_type_check. *)
+           else raise(err)
```

Comment:
1. `if bool then foo else foo`? :thinking: 
2. I don't understand the comment
3. I went and had a look at what the code looks like in merlin, in case there's a difference, and there is, the code there is the following:
    ```ocaml
       match Ctype.closed_type_decl decl with
         Some ty ->
          if not (Msupport.erroneous_type_check ty) then
            raise(Error(sdecl.ptype_loc, Unbound_type_var(ty,decl)))
    ```
   That is: if `ty` is known to already be related to an error, do not complain about it.
   More precisely in this case: if `ty` has been synthesized (e.g. because `Typetexp` failed to "translate" the manifest) do not complain that it is unbound.

Reply:
Yes, I had a small confustion. Should be fixed now.